static/src/javascripts/projects/common/modules/identity/sign-in-gate/helper.js

```diff
@@ -61,6 +81,32 @@ export const hasUserDismissedGate: ({
     return !!prefs[`${name ? `${name}-` : ''}${variant}`];
 };
 
+// check if the user has dismissed the gate within a given timeframe
+export const hasUserDismissedGateInWindow: ({
+    window: DismissalWindow,
+    name?: string,
+    variant: string,
+    componentName: string,
+}) => boolean = ({ window, name = '', componentName, variant }) => {
+    const prefs = userPrefs.get(componentName) || {};
+    const dismissalTZ = Date.parse(
+        prefs[`${name ? `${name}-` : ''}${variant}`]
+    );
+
+    const dismissalWindows = {
+        day: 24,
+        dev: 0.05, // 3 min for testing
+    };
+    const hours = Math.abs(Date.now() - dismissalTZ) / 36e5; //  36e5 is the scientific notation for 60*60*1000, which converts the milliseconds difference into hours.
```

Comment:
I'm a bit confused here, why is the `Math.abs` required?

As the dismissalTZ should always be smaller than the current date, it shouldn't need to be converted to an absolute number. Unless my understanding is wrong. 🤔

Reply:
yes, you are right. I was reading up on TZs and this formula does not make that assumption for any two diffferent TZs but in our case dismissalTZ would always be smaller than now.